# Wednesday Exercises: Variables, Data Types & Basic Operations

## Setup: Run the cell below to define the Olist variables

Execute this code before starting the exercises.

In [ ]:
# Olist variables - run this cell first
product_id = "4244733e06e7ecb4970a6e2683c13e61"
seller_city = "campinas"
price = 58.90
freight = 13.29
review_score = 4
total_orders_brazil = 99441
sp_orders = 41746
order_id = "e481f51cbdc54678b7cc49136f2d6af7"

## Exercises

Complete each task in the code cell below.

1. **Easy:** Print the data types of `product_id`, `price`, and `freight`.

In [2]:
# Your code here
print(type(product_id))
print(type(price))
print(type(freight))

<class 'str'>
<class 'float'>
<class 'float'>


2. **Easy:** Calculate the total order value by adding `price` and `freight`. Store the result in a variable `total_cost` and print it.

In [3]:
# Your code here
total_cost = price + freight
print(total_cost)

72.19


3. **Easy:** A product sold for `price = 58.90` and cost `40.00`. Calculate the profit margin as a percentage: `(revenue - cost) / revenue * 100`. Round to one decimal place and print: `Profit margin: XX.X%` (use f‑string).

In [4]:
# Your code here
profit_margin = (price - 40) / price * 100
print(f"Profit margin: {profit_margin:.1f}%")

Profit margin: 32.1%


4. **Easy:** Perform floor division and modulus on `10` and `3`. Print the results with explanations, e.g., `10 // 3 = 3` and `10 % 3 = 1`.

In [5]:
# Your code here
print("10 // 3 = 3")
print("10 % 3 = 1")

10 // 3 = 3
10 % 3 = 1


5. **Easy:** Using the `seller_city` string, print it in uppercase, title case, and its length, each on a new line.

In [6]:
# Your code here
print(seller_city.upper())
print(seller_city.title())
print(len(seller_city))

CAMPINAS
Campinas
8


6. **Medium:** Extract different slices from `order_id` (`"e481f51cbdc54678b7cc49136f2d6af7"`). Print the first 8 characters, the last 4 characters, and the slice from index 8 to 16 (inclusive start, exclusive end). Label each output.

In [7]:
# Your code here
print(order_id[:8])
print(order_id[-4:])
print(order_id[8:17])

e481f51c
6af7
bdc54678b


7. **Medium:** Create an f‑string that uses `sp_orders` (41,746) and `total_orders_brazil` (99,441) to display: `"SP accounts for 41.98% of Brazil's orders"`. Format the percentage to two decimal places. Hint: calculate `percentage = sp_orders / total_orders_brazil * 100`.

In [8]:
# Your code here
percentage = sp_orders / total_orders_brazil * 100
print(f"SP accounts for {percentage:.2f}% of Brazil's orders")

SP accounts for 41.98% of Brazil's orders


8. **Medium:** Using `price` and `freight`, build an f‑string that says: `"The total order value is R$72.19"`. Ensure freight is included in the total and the value shows two decimal places.

In [9]:
# Your code here
total_order_value = price + freight
print(f"The total order value is R${total_order_value:.2f}")

The total order value is R$72.19


9. **Challenge:** Write a function `format_order(order_id, price, freight)` that returns a string. The string should show the first 8 characters of `order_id` followed by `"..."`, then the total cost (`price + freight`) formatted to two decimals, and the freight value. Example: `"Order 4244733e... costs R$72.19 including R$13.29 freight"`. Test your function with the given `product_id`, `price`, and `freight` and print the result.

In [13]:
def format_order(order_id, price, freight):
  order_str = order_id[:8] + "..."
  total_cost = price + freight
  return f"Order {order_str} costs R${total_cost:.2f} including R${freight:.2f} freight"


